### PR TITLE
Adding product page / Added basic routing / Added custom signup hook / Added Loader

### PR DESCRIPTION
**Pull request by b0llu**
## What to check
- [ ] src/App.js
- [ ] src/Components/Loader/Loader.jsx
- [ ] src/Hook/useSignUp.js
- [ ] src/Pages/ProductPage/ProductComponents/CardContainer/CardContainer.jsx
- [ ] src/Pages/ProductPage/ProductComponents/SectionContainer/SectionContainer.jsx
- [ ] src/Pages/ProductPage/ProductPage.jsx
- [ ] src/context/useReducer.context.js
- [ ] src/reducer/reducer.js
- [ ] src/index.js

## What not to check
- [ ] Do not check any other files

You can preview the final site here,
https://stormpoint-shop.netlify.app/

### CODE REVIEW

File: src/Components/Header/Header.jsx [hunk 1]
```diff
@@ -1,48 +1,56 @@
...
             </div>
             <div className="badge">
-              <Link to="/Wishlist" element={<WishlistPage />}>
-                <i className="far fa-heart"></i>
-              </Link>
-              <div className="number">{forLength.wishlistLength}</div>
+              {/* <Link to="/Wishlist" element={<WishlistPage />}> */}
+              <i className="far fa-heart"></i>
+              {/* </Link> */}
+              <div className="number">1</div>
```
Comment (anik31) [inline src/Components/Header/Header.jsx:37]: suggestion - remove commented code

File: src/Components/Loader/Loader.jsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import './Loader.css'
+
+export const Loader = () => {
+    return (
+        <div className='circle'></div>
+    )
```
Comment (anik31) [inline src/Components/Loader/Loader.jsx:6]: appreciation - good that you are showing loader

File: src/Hook/useSignUp.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      lastName: "Samant",
+      email: "samantdhruv@gmail.com",
+      password: "9130609806",
+    });
+    // saving the encodedToken in the localStorage
+    localStorage.setItem("token", response.data.encodedToken);
+  } catch (error) {
+    console.log(error);
+  }
+};
```
Comment (anik31) [inline src/Hook/useSignUp.js:16]: appreciation - nice work that you have added a custom hook for signup

**Review by anik31 — COMMENTED** [review]
Great work 💯

File: src/Components/Header/Header.jsx (hunk 1 above)
Comment (b0llu) [reply src/Components/Header/Header.jsx:37]: thanks for pointing it out as i have created this site in other repo few things are commented out. these will be fixed soon

File: src/Hook/useSignUp.js (hunk 3 above)
Comment (b0llu) [reply src/Hook/useSignUp.js:16]: thanks!

File: src/Components/Loader/Loader.jsx (hunk 2 above)
Comment (b0llu) [reply src/Components/Loader/Loader.jsx:6]: Thanks!

**Review by Dev-AkshatGupta — APPROVED** [review]
Nice error handling in reducer Naming should be better though
everything else seems great to me

**Pull request merged**